internal/cortex/frontend/transport/handler.go

```diff
@@ -182,10 +171,9 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		writeError(w, err)
 		queryString = f.parseRequestQueryString(r, buf)
 
-		// Check if caching is enabled.
-		if f.lruCache != nil {
-			f.updateFailedQueryCache(err, queryExpressionNormalized, queryExpressionRangeLength, r)
-		}
+		// Try to update cache.
+		_, message := f.failedQueryCache.CallUpdateFailedQueryCache(err, queryExpressionNormalized, queryExpressionRangeLength)
+		level.Debug(util_log.WithContext(r.Context(), f.log)).Log(message)
```

Comment:
Log info is it's cached to help answer any question "why does my query keep being banned?"
```
if cached {
  level.Info(
```

Reply:
I have added back the logic where if it is successful, I log it as info and if it is not successful, I log it as debug. Should I remove the debug log entirely?